students/dave_arasim/session03/mailroom.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+'''Mailroom Part 1 for Session 3, Python 210
+Written by David K. Arasim - 10/22/18'''
+
+#Imported modules
+import os
+
+#Global variables
+donor_db = [("Bill Gates", [653772.32, 12.17]),
+            ("Jeff Bezos", [877.33]),
+            ("Paul Allen", [663.23, 43.87, 1.32]),
+            ("Mark Zuckerberg", [1663.23, 4300.87, 10432.00]),
+            ("Richard Branson", [20000.00, 25000.00, 42.34])
+            ]
+
+#############################################################################################
+#Function section
+
+def main():
+    quit_option = False
+
+    while not(quit_option):
+        os.system('clear')
+        print ('*'*10,'  Main Menu  ','*'*10)
+        print()
+        print('1) Send a Thank You')
+        print('2) Create a Report')
+        print('3) Quit')
+        print()
+        print('Choose an option: ', end='')
+
+        user_choice = input()
+        print()
+
+        try:
+            user_choice = int(user_choice)
+        except ValueError:
+            user_choice = ''
+
+        if user_choice == '':
+            print('Sorry, but that is not a number')
+        else:
+            if   user_choice == 1: send_thank_you()
+            elif user_choice == 2: create_report()
+            elif user_choice == 3: quit_option = True
+            else:
+                print('That option: ', user_choice, ' is out of range', sep='')
+
+        print()
+        if quit_option:
+            print('Quitting process...')
+        else:
+            print('<cr> to continue... ', end='')
+            cont_input = input()  #cont_input for pause (variable not used)
+
+def send_thank_you():
+    quit_thank_you = False
+    os.system('clear')
+
+    while not(quit_thank_you):
+        print("Enter Donor's name, type 'list' for existing Donors (<cr> to exit): ", end='')
+
+        thank_you_choice = input()
+        print()
+
+        if   thank_you_choice == '':     quit_thank_you = True
+        elif thank_you_choice == 'list': show_donor_list()
+        else:
+            find_donor(thank_you_choice)
+            quit_thank_you = True
+
+def show_donor_list():
+    donor_list   = 'List of Donors'
+    donor_list_w = 26
+
+    #Donor list header
+    list_nom = (donor_list, donor_list_w)
+    print('{:{}}'.format(*list_nom))
```

Comment:
Hmm... quite interesting... I never knew a format string can have nested `{}` units! Interesting to see this working, but I think you can just specify the width in the format string (`'{:26}'.format(donor_list)`). I suppose you wanted to be able to avoid hard-coding `26` in multiple places, and if that was the intention and if this is the way to implement that, kudos! And thanks for teaching me something new and interesting!